src/pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index.jsx

```diff
@@ -80,14 +79,49 @@ const MarkerHeatmap = ({ experimentId }) => {
       .configureEmbedding?.clusteringSettings.methodSettings.louvain.resolution,
   ) || false;
 
+  const groupedCellSets = useSelector((state) => {
+    if (!config?.groupedTracks) return undefined;
+
+    const groupedCellClasses = getCellSetsHierarchyByKeys(config.groupedTracks)(state);
+    return groupedCellClasses.map((cellClass) => cellClass.children).flat();
+  }, _.isEqual);
+
   useEffect(() => {
     if (!louvainClustersResolution) dispatch(loadProcessingSettings(experimentId));
     if (!config) dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
     if (!hierarchy?.length) dispatch(loadCellSets(experimentId));
   }, []);
 
-  const updatePlotWithChanges = (updatedField) => {
-    dispatch(updatePlotConfig(plotUuid, updatedField));
+  const userUpdatedPlotWithChanges = (userUpdatedField) => {
```

Comment:
Wouldn't it be more clear if instead of this wrapper to still use `updatePlotWithChanges`. But these ifs checking what was updated to be instead three `useEffects` ? Something like
```
useEffect(()=>{
    dispatch(updatePlotConfig(plotUuid, {
        selectedTracks: [config.selectedCellSet],
        groupedTracks: [config.selectedCellSet],
      };));
},[config.selectedCellSet])
```

Reply:
The same for `updatesToDispatch.nMarkerGenes` and `updatesToDispatch.selectedGenes`

Reply:
`userUpdatedPlotWithChanges ` is just a rename of `updatePlotWithChanges`, we're still using it

Reply:
The thing is `userUpdatedPlotWithChanges` isn't being triggered by a `useEffect`, it's triggered by the `onUpdate` of different components

Reply:
Just mentioned that because it was a bit unclear, maybe a comment explaining why this is needed would be nice. For me the useEffects are clearer since you know when something changes that this will be executed.